tasks/icinga2-node-config.yml

```diff
@@ -0,0 +1,63 @@
+# This file does not run on master nodes
+---
+- name: Check if Certificate Authority is available
+  become: yes
+  stat:
+    path: "{{ i2_pki_path }}/trusted_parent.crt"
+  register: trusted_authority
+
+- name: Initialize Certificate Directory
+  become: yes
+  file:
+    path: "{{ i2_pki_path }}"
+    state: directory
+    owner: "{{ i2_user }}"
+    group: "{{ i2_group }}"
+    mode: 0700
+
+- name: Generate Client Ticket on Master
+  become: yes
+  command: icinga2 pki ticket --cn {{ ansible_fqdn }}
+  register: pki_ticket
+  delegate_to: "{{ groups['masters'][0] }}"
+  when: client_certificate.stat.exists is sameas false
+
+  # Todo: How can / will this work for nodes unable to connect
+  # directly to master?
+- name: Retrieve Public Certificate From Master
+  become: yes
+  command: >
+    icinga2 pki save-cert --key {{ i2_pki_path }}/{{ ansible_fqdn }}.key
+      --cert {{ i2_pki_file }}
+      --trustedcert {{ i2_pki_path }}/trusted_parent.crt
+      --host {{ hostvars[groups['masters'][0]]['ansible_fqdn'] }}
+  when: trusted_authority.stat.exists is sameas false
+
+- name: Create Host Certificate and set-up Node (Top-Down)
```

Comment:
Maybe the entire set-up and PKI should be handled by ansible rather than using the node setup.
That would make the `icinga2-config.yml` also simpler because we wouldn't have to skip some steps in some cases (e.g. where the client-certificates haven't been generated yet).